src/main/java/com/rapid7/client/dcerpc/msvcctl/messages/RCreateServiceWRequest.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017, Rapid7, Inc.
+ *
+ * License: BSD-3-clause
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ *  Redistributions in binary form must reproduce the above copyright
+ * notice, this list of conditions and the following disclaimer in the
+ * documentation and/or other materials provided with the distribution.
+ *
+ *  Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ *
+ */
+
+package com.rapid7.client.dcerpc.msvcctl.messages;
+
+import com.rapid7.client.dcerpc.io.PacketOutput;
+import com.rapid7.client.dcerpc.messages.RequestCall;
+import com.rapid7.client.dcerpc.msvcctl.enums.ServiceManagerAccessLevel;
+import com.rapid7.client.dcerpc.msvcctl.objects.IServiceConfigInfo;
+import com.rapid7.client.dcerpc.objects.ContextHandle;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+public class RCreateServiceWRequest extends RequestCall<RCreateServiceWResponse> {
+    private final static short OP_NUM = 12;
+    private ContextHandle serviceHandle;
+    private IServiceConfigInfo serviceConfigInfo;
+    private int access;
+    private String serviceName;
+
+    public RCreateServiceWRequest(
+        final ContextHandle handle,
+        final IServiceConfigInfo serviceConfigInfo,
+        final ServiceManagerAccessLevel managerAccessLevel,
+        final String serviceName){
+        super(OP_NUM);
+        this.serviceHandle = handle;
+        this.serviceConfigInfo = serviceConfigInfo;
+        this.access = managerAccessLevel.getValue();
+        this.serviceName = serviceName;
+    }
+
+    @Override
+    public void marshal(PacketOutput packetOut) throws IOException {
+        packetOut.write(serviceHandle.getBytes());
+        if (serviceName!= null)
+            packetOut.writeString(serviceName, true);
+        else packetOut.writeNull();
```

Comment:
According to https://msdn.microsoft.com/en-us/library/cc245923.aspx ; the serviceName MUST not be NULL.

Reply:
^ Last thing left to be addressed. Please push PR with updates

Reply:
This has been addressed locally.